### DISCUSSION TITLE
How to sync a clone of this repo after changing the name and adding more code.

### DISCUSSION THREAD

**Discussion by yorsant** [nickjj/docker-rails-example#40] [General]
I have tried to add this repo as an upstream to my clone of it and merge the main branch, but after changing the name for the project and adding some code it's becoming hard to solve all the conflicts there.
Has anyone lived this pain?
Is there any way to make this possible without having too many conflicts?
Is there any other way to keep in sync?

Thanks.

**Comment by nickjj** [comment]
What type of pain are you having?

**Comment by yorsant** [comment]
I'm having too many conflicts. Some of them are due to changing the name of the project, maybe re-running the script could help.
Others are due to updates I did on my clone which caused some files like `gemfile.lock` to be full of difficult conflicts.
I just wanted to know if there is a good practice I should follow. 
Or any advice on how to use this repo as an upstream for my project.

**Reply by nickjj** [reply]
It's expected both projects (your app and this app) will diverge a lot since they're separate apps. Besides depending on git to merge or patch specific files you could use a combination of this repo's changelog to track what changed and manually add them into your project, or scan the list of commits.

At this point in the project the main things that are changing are updating dependencies. There may be other things as well which would make scanning the list of commits from time to time be something you may want to do.

**Comment by JBerendes** [comment]
I was about to ask the same question and found this conversation helpful. My project will require ActiveAdmin causing the project to greatly diverge from this one. My plan will be to keep a synced repo of this running along side the new project and compare the changelog of each repo. And I'll contribute back as I find dependencies out of sync. I'm really glad to have this project as a jump off point, so thank you.

**Reply by nickjj** [reply]
Thanks, no problem. By the way I have an automated script that goes through and updates all of the dependencies and updates the changelog. It's something I run every week or 2.

**Reply by JBerendes** [reply]
Ah. So that makes plenty of sense with keeping the commit messages concise.
